src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -198,22 +205,25 @@ def _osd_flags():
     def list(self):
         return self._osd_flags()
 
-    def bulk_set(self, flags):
-        """
-        The `recovery_deletes`, `sortbitwise` and `pglog_hardlimit` flags cannot be unset.
-        `purged_snapshots` cannot even be set. It is therefore required to at
-        least include those four flags for a successful operation.
-        """
+    def bulk_delete(self, flags):
         assert isinstance(flags, list)
 
-        enabled_flags = set(self._osd_flags())
-        data = set(flags)
-        added = data - enabled_flags
-        removed = enabled_flags - data
-        for flag in added:
-            CephService.send_command('mon', 'osd set', '', key=flag)
-        for flag in removed:
+        unset_flags = flags - self.unsupported_set_unset_flags
+        # `osd unset` is idempotent. If a flag is not present in the map, we
+        # still get a success.
+        for flag in unset_flags:
             CephService.send_command('mon', 'osd unset', '', key=flag)
-        logger.info('Changed OSD flags: added=%s removed=%s', added, removed)
+        logger.info('Changed OSD flags: removed=%s', unset_flags)
+        return sorted(unset_flags)
 
-        return sorted(enabled_flags - removed | added)
+    def bulk_set(self, flags):
```

Comment:
`bulk_set` means`PUT` here, and with these changes it becomes a partial update, which breaks [HTTP 1.1 semantics](https://tools.ietf.org/html/rfc7231#section-4.3.4).

Reply:
See my comment at https://github.com/ceph/ceph/pull/26988/files/f0acec26e408f925cdc01aa3f089b8a9cac9c727#r266561242

I truly don't know what the best approach would be, so I'm obviously open to input.